### ISSUE TITLE
About the real data distributions when calculating the FID 

### ISSUE THREAD

**Issue by tangjilin** [RenYurui/Global-Flow-Local-Attention#34]
Hi. I have a question about the FID.
For FID, why do you calculate the real data distributions over the whole training set rather than the testing set? 
In my opinion, since we have the ground-truth of generated images in the testing set, it is more reasonable to calculate the real data distributions over the testing set to get the FID. 
Thanks!

**Comment by RenYurui** [comment]
The FID metric does not need ground-truth images. It compares the distributions of real and fake images. Therefore, accurate real image distributions can help to obtain accurate evaluation results. We find that calculating the distributions using the training images can obtain more stable results. (since more real images are used to calculate the real distribution)

**Comment by tangjilin** [comment]
Thanks. By the way, I want to know why don't you calculate the masked-FID over the Market Dataset?

**Comment by RenYurui** [comment]
It may be unreasonable to calculate the mask-FID. Images with different backgrounds can have similar distributions.

**Comment by tangjilin** [comment]
Can you explain it in more details？In the paper, you calculate the mask-LPIPS to alleviate the influence of the backgrounds. What's the difference between the masked-FID and the masked-LPIPS?

**Comment by RenYurui** [comment]
LPIPS requires ground-truth images to calculate the reconstruction error. Therefore, the different backgrounds between real and fake images lead to inaccurate evaluation results. FID calculates the distribution distance. It does not require gt images. Please check their papers for more details.

**Comment by tangjilin** [comment]
Got it. Thanks for your reply!
